### ISSUE TITLE
discoverd deployment panic in CI

### ISSUE THREAD

**Issue by titanous** [flynn/flynn#2941]
This appears to be very common on CI (I think it happens on every CI run):

```text
2016/06/17 15:13:12 checking for existing discoverd process at http://10.69.86.6:1111
2016/06/17 15:13:12 discoverd responding at http://10.69.86.6:1111 taking over
2016/06/17 15:13:12 Created deployment
discoverd is exiting
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x52db73]

goroutine 1 [running]:
github.com/flynn/flynn/pkg/shutdown.(*handler).exit(0xc208038150, 0x0, 0x0, 0x0, 0x8f51a0, 0xc20802e050)
	/home/ubuntu/go/src/github.com/flynn/flynn/pkg/shutdown/handler.go:69 +0x12e
github.com/flynn/flynn/pkg/shutdown.Exit()
	/home/ubuntu/go/src/github.com/flynn/flynn/pkg/shutdown/handler.go:40 +0x73
github.com/flynn/flynn/discoverd/deployment.(*Deployment).decode(0xc20809cd20, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/ubuntu/go/src/github.com/flynn/flynn/discoverd/deployment/deployment.go:223 +0x113
github.com/flynn/flynn/discoverd/deployment.(*Deployment).MarkPerforming(0xc20809cd20, 0xc20802f970, 0xf, 0x3c, 0x0, 0x0)
	/home/ubuntu/go/src/github.com/flynn/flynn/discoverd/deployment/deployment.go:91 +0xa0
main.(*Main).Run(0xc2080800e0, 0xc20800a010, 0x6, 0x6, 0x0, 0x0)
	/home/ubuntu/go/src/github.com/flynn/flynn/discoverd/main.go:124 +0xc85
main.main()
	/home/ubuntu/go/src/github.com/flynn/flynn/discoverd/main.go:39 +0x134
```

/cc @josephglanville

**Comment by josephglanville** [comment]
This is an interesting one. When we create the deployment we wait for `EventKindCurrent` before continuing to `MarkPerforming()`. So for `d.meta` to be nil it must mean that we received the current message before receiving a `ServiceMeta` event.

Making `MarkPerforming()` wait until there is metadata that it can decode is probably fine though, we have the timeout in place anyways.

**Comment by titanous** [comment]
Fixed by #2945.
